portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,12 +33,26 @@
 public class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String commentcount = request.getParameter("count");
+    int maxcomments;
+    try {
+        maxcomments = Integer.parseInt(commentcount);
+    } catch(NumberFormatException e){
+        // Display nothing for now.
+        maxcomments = 0;
+    }
     Query query = new Query("Show").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     List<String> shows = new ArrayList<>();
+    int count = 0;
     for (Entity entity : results.asIterable()) {
-      shows.add((String)entity.getProperty("show"));
+      if(count < maxcomments){
+        shows.add((String)entity.getProperty("show"));
+      } else{
```

Comment:
nit: indentation.  Maybe enable a linter to find and fix these automatically? See go/sps-code-reviews

Reply:
Did my best to fix. We aren't suppose to install linters I believe